components/tor/tor_launcher_factory.cc

```diff
@@ -194,6 +196,10 @@ void TorLauncherFactory::SetupBridges(tor::BridgesConfig bridges_config) {
 void TorLauncherFactory::AddObserver(TorLauncherObserver* observer) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   observers_.AddObserver(observer);
+  if (last_init_message_) {
+    observer->OnTorInitializing(last_init_message_->percentage,
```

Comment:
what is this for that we have to notify observer upon adding observer?

Reply:
To send the current connection status. If the connection is in progress and the user opens one more tab, it will display the _disconnected_ status until the next initialization message arrives. There may be a long delay between init messages in some cases (e.g. snowflake bridges enabled).